components/modals/appModals/AppModals.tsx

```diff
@@ -1,84 +1,88 @@
-      this.props.uiCloseEditQuestionModal();
-    } else {
-      this.props.uiCloseAddQuestionModal();
-    }
-  };
+export const AppModals = memo(() => {
+  const addQuestionModalRef = useRef<HTMLDivElement>(null);
+  const addQuestionConfirmationModalRef = useRef<HTMLDivElement>(null);
 
-  render() {
-    return (
-      <React.Fragment>
-        <CSSTransition
-          in={this.props.addQuestionModalState.open}
-          unmountOnExit={true}
-          mountOnEnter={true}
-          classNames={{
-            enter: styles.fadeEnter,
-            enterActive: styles.fadeEnterActive,
-            enterDone: styles.fadeEnterDone,
-            exit: styles.fadeExit,
-            exitActive: styles.fadeExitActive,
-          }}
-          timeout={timeout}
-        >
-          <AddQuestionModal
-            originalQuestion={this.props.addQuestionModalState.data}
-            onClose={this.closeQuestionModal}
-          />
-        </CSSTransition>
+  const dispatch = useDispatch();
+  const addQuestionModalState = useSelector((state) => state.ui.addQuestionModal);
+  const isAddQuestionConfirmationModalOpen = useSelector(
+    (state) => state.ui.isAddQuestionConfirmationModalOpen
+  );
 
-        <CSSTransition
-          in={this.props.isAddQuestionConfirmationModalOpen}
-          unmountOnExit={true}
-          mountOnEnter={true}
-          classNames={{
-            enter: styles.fadeEnter,
-            enterActive: styles.fadeEnterActive,
-            enterDone: styles.fadeEnterDone,
-            exit: styles.fadeExit,
-            exitActive: styles.fadeExitActive,
-          }}
-          timeout={timeout}
-        >
-          <AddQuestionConfirmationModal onClose={this.props.uiCloseAddQuestionConfirmationModal} />
-        </CSSTransition>
-      </React.Fragment>
-    );
-  }
-}
+  const closeQuestionModal: CommonModalProps['onClose'] = useCallback(
+    (args) => {
+      if (addQuestionModalState.onClose) {
+        addQuestionModalState.onClose(args);
+      }
+      if (addQuestionModalState.data) {
+        dispatch(ActionCreators.uiCloseEditQuestionModal());
+      } else {
+        dispatch(ActionCreators.uiCloseAddQuestionModal());
+      }
+    },
+    [addQuestionModalState, dispatch]
+  );
 
-const mapStateToProps = (state: AppState) => {
-  return {
-    addQuestionModalState: state.ui.addQuestionModal,
-    isAddQuestionConfirmationModalOpen: state.ui.isAddQuestionConfirmationModalOpen,
-  };
-};
+  const closeConfirmationModal: CommonModalProps['onClose'] = useCallback(
+    (_args) => {
+      dispatch(ActionCreators.uiCloseAddQuestionConfirmationModal());
+    },
+    [dispatch]
+  );
 
-const mapDispatchToProps = {
-  uiCloseAddQuestionModal: ActionCreators.uiCloseAddQuestionModal,
-  uiCloseEditQuestionModal: ActionCreators.uiCloseEditQuestionModal,
-  uiCloseAddQuestionConfirmationModal: ActionCreators.uiCloseAddQuestionConfirmationModal,
-};
+  return (
+    <React.Fragment>
```

Comment:
Maybe you should change React.Fragment to empty tags, because you don't use "key" in it.

```
<>
{...}
</>
```